tls/s2n_cipher_suites.c

```diff
@@ -1227,15 +1228,27 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                 continue;
             }
 
+            /* For TLS1.3 when PSKs are present, the server must consider the hash algorithm associated with the chosen PSK,
+             * when choosing a cipher suite. Server MUST reject any cipher suite without a matching hash algorithm and 
+             * continue to the next candidate. 
+             * */     
+            if (conn->actual_protocol_version >= S2N_TLS13 && conn->psk_params.chosen_psk != NULL) {
```

Comment:
You probably don't need the protocol check here. PSKs are only used in TLS1.3. If you do keep it, write a test for it.